src/Components/MovieModal.js

```diff
@@ -1,59 +1,91 @@
 import { updateMovie, addMovie, deleteMovie } from '../Actions/actions'
 
+const validate = values => {
+    const errors = {};
+    if (!values.title) {
+        errors.title = 'Required';
+    } else if (values.title.length > 20) {
+        errors.title = 'Must be 20 characters or less';
+    }
+
+    if (!values.release_date) {
+        errors.release_date = 'Required';
+    } else if (!/^([0-9]{2})-([0-9]{2})-([0-9]{4})$/.test(values.release_date)) {
+        errors.release_date = 'Date format is DD/MM/YYYY';
+    }
+
+    if (!values.poster_path) {
+        errors.poster_path = 'Required';
+    } else if (!/((https?):\/\/)?(www.)?[a-z0-9]+(\.[a-z]{2,}){1,3}(#?\/?[a-zA-Z0-9#]+)*\/?(\?[a-zA-Z0-9-_]+=[a-zA-Z0-9-%]+&?)?$/.test(values.poster_path)) {
+        errors.poster_path = 'Invalid poster path';
+    }
+
+    if (!values.genres) {
+        errors.genres = 'Required';
+    } else if (values.genres.length > 20) {
+        errors.genres = 'Must be 20 characters or less';
+    }
+
+    if (!values.overview) {
+        errors.overview = 'Required';
+    } else if (values.overview.length > 150) {
+        errors.overview = 'Must be 150 characters or less';
+    }
+
+    if (!values.runtime) {
+        errors.runtime = 'Required';
+    } else if (typeof values.runtime !== "number") {
+        errors.runtime = 'Only numbers are allowed';
+    }
+    return errors;
+};
+
 const MovieModal = ({ movie, text, isEdit, isDelete, toggleMovieModal, updateMovie, addMovie }) => {
-    const [title, setTitle] = useState('');
-    const [release_date, setRelease_date] = useState('');
-    const [poster_path, setPoster_path] = useState('');
-    const [genres, setGenres] = useState('');
-    const [overview, setOverview] = useState('');
-    const [runtime, setRuntime] = useState('');
-    const [id, setMovieId] = useState(null);
+    const form = useFormik({
+        initialValues: {
+            title: '',
+            release_date: '',
+            poster_path: '',
+            genres: '',
+            overview: '',
+            runtime: '',
+            id: '',
+        },
+        validate,
+        onSubmit: values => {
+            isEdit ? updateMovie({...movie, ...values}) : addMovie(values);
+            toggleMovieModal();
+        },
+    });
 
     useEffect(() => {
         if (isEdit) {
-            setTitle(movie.title);
-            setRelease_date(movie.release_date);
-            setPoster_path(movie.poster_path);
-            setGenres(movie.genres);
-            setOverview(movie.overview);
-            setRuntime(movie.runtime);
-            setMovieId(movie.id)
+            form.values.title = movie.title;
+            form.values.release_date = movie.release_date;
+            form.values.poster_path = movie.poster_path;
+            form.values.genres = movie.genres;
+            form.values.overview = movie.overview;
+            form.values.runtime = movie.runtime;
+            form.values.id = movie.id;
         }
     }, [movie])
```

Comment:
I'm not sure, but maybe you can use `enableReinitialize` here
```js
    const form = useFormik({
        enableReinitialize: true,
        initialValues: {
            title: isEdit ? movie.title : '',
            release_date: isEdit ? movie.release_date : '',
            poster_path: isEdit ? movie.poster_path : '',
            genres: isEdit ? movie.genres : '',
            overview: isEdit ? movie.overview : '',
            runtime: isEdit ? movie.runtime : '',
            id: isEdit ? movie.id : '',
        },
        validate,
        onSubmit: values => {
            isEdit ? updateMovie({...movie, ...values}) : addMovie(values);
            toggleMovieModal();
        },
    });
```

and you can remove `useEffect`